plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/edit.tsx

```diff
@@ -19,20 +22,50 @@ export default function AddToCartWithOptionsGroupedProductSelectorEdit(
 	props: BlockEditProps< Attributes >
 ) {
 	const { className } = props.attributes;
-	const { current: currentProductType } = useProductTypeSelector();
 	const { product } = useProductDataContext();
-	const productType =
-		product.id === 0 ? currentProductType?.slug : product.type;
-
 	const blockProps = useBlockProps( {
 		className,
 	} );
 	const innerBlocksProps = useInnerBlocksProps( blockProps, {
 		template: GROUPED_PRODUCT_ITEM_TEMPLATE,
 		templateLock: 'all',
 	} );
+	const [ groupedProduct, setGroupedProduct ] = useState<
+		ProductResponseItem[]
+	>( [] );
+
+	// If product ID is 0, then we must be editing a template.
+	// Fetch an existing grouped product so template can be edited.
+	useEffect( () => {
+		if ( product.id !== 0 && product.type === 'grouped' ) {
+			setGroupedProduct( [ product as ProductResponseItem ] );
+		}
+		if ( groupedProduct.length === 0 && product.id === 0 ) {
+			resolveSelect( productsStore )
+				.getProducts( {
+					type: 'grouped',
+					per_page: 1,
+				} )
+				.then( ( fetchedGroupedProduct ) => {
+					if ( fetchedGroupedProduct.length > 0 ) {
+						setGroupedProduct( fetchedGroupedProduct );
+					} else {
+						// If there are no grouped products, query for any other product type.
+						resolveSelect( productsStore )
+							.getProducts( {
+								per_page: 1,
+							} )
+							.then( ( fetchedProduct ) => {
+								if ( fetchedProduct.length > 0 ) {
+									setGroupedProduct( fetchedProduct );
+								}
+							} );
+					}
+				} );
+		}
+	}, [ groupedProduct, product ] );
```

Comment:
I'm not sure to understand what is `groupedProduct` used for? Apparently it's only used in the `if` below, but it's not clear to me what's the benefit of the change. :thinking:

Reply:
I added this `groupedProduct` to match the product context in `ProductItemTemplateEdit`. Both of these are used to set the context based on whether an appropriate grouped product has been provided, and if not, look for an existing grouped product or fallback to any other type of product.

I thought it was best to keep these contexts the same, as if there is no `groupedProduct`, the inner blocks will not be rendered. 

Let me know if you think there's a better way to do this.

Reply:
It makes sense, but what are the benefits of doing so instead of simply rendering the wrapping `<div>` here and letting the inner blocks (in this case `ProductItemTemplateEdit`) handle whether they should render or not?

I don't have a strong opinion, but I feel that would allow simplifying the logic a lot? :thinking:

Reply:
Oh I see, yes that would be much simpler! I've removed all the product context from the selector in https://github.com/woocommerce/woocommerce/pull/57690/commits/c18e4bdbaa94b2d60c20a89b56ab953a69bf472e. I guess I was trying to maintain the existing logic but with additional product context, but yes, I don't think this is necessary 😄